mobify/progressive-web-scaffold
# Disable service worker from Astro apps

Due to a mish-mash between the caching by the service worker and how we produce preview on Astro, the preview experience on Android is currently broken if the service worker is involved. We 'd like to disable the service worker if MP is running inside an Astro app.

 **Linked PRs**: https://github.com/mobify/progressive-web-scaffold/pull/236

## Changes
- Disable the service worker if running in an Astro app

## How to test-drive this PR
- The `service-worker-loader.js` file is currently always loaded from `merlinspotions.com` so to test this change https://github.com/mobify/progressive-web-scaffold/blob/develop/web/app/loader.js#L19 to be `const SW_LOADER_PATH = https://localhost:8443/service-worker-loader.js?preview=${IS_PREVIEW}&b=${cacheHashManifest.buildDate}`
- Run `npm run dev`
- Preview [Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fwww.merlinspotions.com%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- From the web, the service worker should be loaded and working
- From an Android Astro app, the service worker not be there
